packages/expo-router/src/modal/web/modalStyles.ts

```diff
@@ -0,0 +1,34 @@
+// NOTE: This file intentionally wraps the CSS-module import so that Fast Refresh
+// does NOT treat the generated stub as part of the React boundary and cause
+// a full refresh on web.
+//
+// Why?
+// -----
+// • Metro recompiles *every* direct dependency of the file you edit. When you
+//   change a component that imports a CSS-module, Metro re-emits the JS stub
+//   that represents that `*.module.css` file, marking it "updated".
+// • The stub exports a plain object (class-name strings) which React-Refresh
+//   does **not** recognise as a component -> the stub itself cannot be a refresh
+//   boundary.
+// • If an UPDATED, NON-BOUNDARY module bubbles all the way up the dependency
+//   graph without meeting another UPDATED boundary, React-Refresh falls
+//   back to a full reload — wiping React state and breaking fast-refresh.
+// • That is exactly what happened: `modal.module.css` changed, its parent
+//   (ModalStack.web.tsx) *didn't* change, so refresh bailed out.
+//
+// Solution
+// ---------
+// We import the CSS here and immediately call `module.hot.accept()` so the
+// update is consumed at this level. React-Refresh no longer has to walk past
+// the stub, the rest of the graph (including the edited component) hot-swaps
+// normally, and local state is preserved.
+
+import styles from '../../../assets/modal.module.css';
+
+// Swallow future updates to the CSS stub so they don't bubble up and force a
+// full refresh. `module.hot` is provided by Metro in development builds.
+
+// @ts-expect-error — `module` is injected at runtime by Metro.
+if (module.hot) module.hot.accept();
+
+export default styles;
```

Comment:
Yes, simply importing a CSS module fully breaks fast refresh. I had to go deep down the rabbit hole to find the cause of the failing tests and state preserving. The explanation is in that file’s comments. Just calling module.hot.accept() was enough to fix it, pretty similar to what Webpack does.